src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -211,6 +223,51 @@ export default class DeployFromSettingsController {
   }
 
   /**
+   * Displays new secret creation dialog.
+   *
+   * @param {!angular.Scope.Event} event
+   * @export
+   */
+  handleCreateSecretDialog(event) {
+    showCreateSecretDialog(this.mdDialog_, event, this.namespace)
+        .then(
+            /**
+             * Handles create secret dialog result. If the secret was created successfully, then it
+             * will be selected,
+             * otherwise None is selected.
+             * @param {string|undefined} response
+             */
+            (response) => {
+              if (response) {
+                this.imagePullSecret = response;
+                this.secrets = this.secrets.concat(this.imagePullSecret);
+              } else {
+                this.imagePullSecret = '';
+              }
+            },
+            () => { this.imagePullSecret = ''; });
+  }
+
+  /**
+   * Queries all secrets for the given namespace.
+   * @param {string} namespace
+   * @export
+   */
+  getSecrets(namespace) {
+    /** @type {!angular.Resource<!backendApi.SecretsList>} */
+    let resource = this.resource_(`api/secrets/${namespace}`);
+    /** @type {!angular.$q.Promise} */
+    let promise = resource.get().$promise;
+    promise.then((res) => { this.secrets = res.secrets; });
```

Comment:
Please $log when there is and error loading secrets. This would be useful when later displaying error messages to the user.